crates/bevy_math/src/compass.rs

```diff
@@ -72,6 +113,57 @@ pub enum CompassOctant {
     NorthWest,
 }
 
+impl CompassOctant {
+    /// Converts a standard index to a [`CompassOctant`].
+    ///
+    /// Starts at 0 for [`CompassOctant::North`] and increments clockwise.
+    pub const fn from_index(index: usize) -> Option<Self> {
+        match index {
+            0 => Some(Self::North),
+            1 => Some(Self::NorthEast),
+            2 => Some(Self::East),
+            3 => Some(Self::SouthEast),
+            4 => Some(Self::South),
+            5 => Some(Self::SouthWest),
+            6 => Some(Self::West),
+            7 => Some(Self::NorthWest),
+            _ => None,
+        }
+    }
+
+    /// Converts a [`CompassOctant`] to a standard index.
+    ///
+    /// Starts at 0 for [`CompassOctant::North`] and increments clockwise.
+    pub const fn to_index(self) -> usize {
+        match self {
+            Self::North => 0,
```

Comment:
Random thought I had, (this may be irrelevant or out of scope for this change) - is it worth foregoing the nice index integer incrementing and instead making the cardinal directions match up between `CompassOctant` and `CompassQuadrant`? I'm thinking of a scenario where the integer data is serialized & conversion between the two becomes a bit frustrating for the user of the API.

Reply:
I think it's better to just have a `From`+`TryFrom` impls for the conversions, the indices are better to have a reasonable order, because they'll be used as indices to arrays.

Reply:
Would you like me to add From + TryFrom traits too? I'd like to keep the existing methods since they're const.

Reply:
I'm fine either way - I'll defer to your judgement here. I think I was partially confused too and thought that the `NavNeighbors` `get/set` took a `CompassQuadrant` as well - though I do think it's a potential pitfall that `NavNeighbors.neighbors[CompassOctant::East.to_index()] != NavNeighbors.neighbors[CompassQuadrant::East.to_index()]`

Reply:
Yeah, I think that the strongly typed helper methods are going to be preferred so we should be safe.